apps/prairielearn/src/tests/permissions/effectiveUser.test.js

```diff
@@ -424,8 +483,77 @@ describe('effective user', function () {
         cookie:
           'pl_test_user=test_instructor; pl_access_as_administrator=inactive; pl_requested_course_role=None',
       };
-      const response = await helperClient.fetchCheerio(context.pageUrlCourseInstance, { headers });
+      const response = await helperClient.fetchCheerio(context.pageUrlTestCourseInstance, {
+        headers,
+      });
       assert.isTrue(response.ok);
     },
   );
+
+  step('reset instructor course role to maximum permissions', async () => {
+    await sqldb.callAsync('course_permissions_update_role', [1, instructorId, 'Owner', 1]);
+    await sqldb.callOneRowAsync('course_instance_permissions_update_role', [
+      1,
+      instructorId,
+      1,
+      'Student Data Editor',
+      2,
+    ]);
+  });
+
+  step(
+    'instructor can access their own course when requesting access as institution administrator',
+    async () => {
+      const headers = {
+        cookie:
+          'pl_test_user=test_instructor; pl_access_as_administrator=inactive; pl_requested_uid=institution-admin@illinois.edu',
+      };
+      const response = await helperClient.fetchCheerio(context.pageUrlTestCourse, {
+        headers,
+      });
+      assert.isTrue(response.ok);
+    },
+  );
+
+  step(
+    'instructor can access their own course instance when requesting access as institution administrator',
+    async () => {
+      const headers = {
+        cookie:
+          'pl_test_user=test_instructor; pl_access_as_administrator=inactive; pl_requested_uid=institution-admin@illinois.edu',
+      };
+      const response = await helperClient.fetchCheerio(context.pageUrlTestCourseInstance, {
+        headers,
+      });
+      assert.isTrue(response.ok);
+    },
+  );
```

Comment:
Institution admins implicitly have course Owner and course instance Student Data Editor permissions, which means that any instructor who does not have maximum permissions will fail to request access as an institution admin. If we happen to have a situation where an institution admin is enrolled in a course and a less-privileged instructor wants to emulate them, I think instructors can just deal with that 😬

I should probably add tests for the above behavior; I know it works because these tests failed before I added the `reset instructor course role to maximum permissions` step above.

Reply:
I added a test case for this in 8a53407.